src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/BookingServiceTest.kt

```diff
@@ -4302,6 +4328,486 @@ class BookingServiceTest {
+        .withApplication(application)
+        .produce()
+
+      every { mockBedRepository.findByIdOrNull(bed.id) } returns bed
+
+      every { mockBookingRepository.save(any()) } answers { it.invocation.args[0] as BookingEntity }
+
+      every { mockBookingRepository.findByBedIdAndArrivingBeforeDate(bed.id, departureDate, null) } returns listOf()
+      every {
+        mockLostBedsRepository.findByBedIdAndOverlappingDate(
+          bed.id,
+          arrivalDate,
+          departureDate,
+          null,
+        )
+      } returns listOf()
+      every { mockAssessmentRepository.findByIdOrNull(application.id) } returns assessment
+
+      every { mockTurnaroundRepository.save(any()) } answers { it.invocation.args[0] as TurnaroundEntity }
+
+      every { mockWorkingDayCountService.addWorkingDays(any(), any()) } answers { it.invocation.args[0] as LocalDate }
+
+      every { mockCas3DomainEventService.saveBookingProvisionallyMadeEvent(any()) } just Runs
+
+      every { mockAssessmentService.closeAssessment(user, application.id) } returns AuthorisableActionResult.Success(
+        ValidatableActionResult.FieldValidationError(ValidationErrors()),
+      )
+
+      mockkStatic(Sentry::class)
+      every { Sentry.captureException(any()) } returns SentryId.EMPTY_ID
+
+      val authorisableResult = bookingService.createTemporaryAccommodationBooking(
+        user,
+        premises,
+        crn,
+        "NOMS123",
+        arrivalDate,
+        departureDate,
+        bed.id,
+        application.id,
+        false,
+      )
+      assertThat(authorisableResult is AuthorisableActionResult.Success).isTrue
+
+      val validatableResult = (authorisableResult as AuthorisableActionResult.Success).entity
+      assertThat(validatableResult is ValidatableActionResult.Success)
+
+      verify(exactly = 1) {
+        mockBookingRepository.save(
+          match {
+            it.crn == crn &&
+              it.premises == premises &&
+              it.arrivalDate == arrivalDate &&
+              it.departureDate == departureDate &&
+              it.application == application &&
+              it.status == BookingStatus.provisional
+          },
+        )
+      }
+
+      verify(exactly = 1) {
+        mockCas3DomainEventService.saveBookingProvisionallyMadeEvent(
+          match {
+            it.crn == crn &&
+              it.premises == premises &&
+              it.arrivalDate == arrivalDate &&
+              it.departureDate == departureDate &&
+              it.application == application
+          },
+        )
+      }
+      verify(exactly = 1) {
+        mockAssessmentService.closeAssessment(user, application.id)
+      }
+      verify(exactly = 1) {
+        Sentry.captureException(any())
+      }
+    }
+
+    @Test
+    fun `createTemporaryAccommodationBooking saves Booking and creates domain event and log error when closing assessment failed with runtime exception`() {
```

Comment:
as mentioned in other comment, i don't think we should stop runtime exceptions

Reply:
Its a requirement to not block the booking flow while cancelling the assessment.